lib/conjur/puppet_module/identity.rb

```diff
@@ -41,10 +44,24 @@ def from_wincred(uri)
 
           require 'wincred/wincred'
 
-          WinCred.enumerate_credentials
-                  .select { |cred| cred[:target].start_with?(uri.to_s) || cred[:target] == uri.host }
-                  .map { |cred| [cred[:username], cred[:value].force_encoding('utf-16le').encode('utf-8')] }
-                  .first
+          Puppet.debug "Finding Conjur credentials in WinCred storage for uri: #{uri}"
+          matching_creds = WinCred.enumerate_credentials.select do |cred|
+            cred[:target].start_with?(uri.to_s) || \
+              cred[:target] == "#{uri.host}:#{uri.port}" || \
```

Comment:
This is definitely an improvement!

1. I worry that this is still brittle since the regex in `init.pp` isn't parsing the URL and might not get `host` or `host+port`. 
2. I'm not sure (right this second) why we need this criteria `cred[:target].start_with?(uri.to_s)`

Reply:
> I worry that this is still brittle since the regex in init.pp isn't parsing the URL and might not get host or host+port.

It's brittle for sure but keep in mind that for this we are explicitly just fixing the HFT usage which has a very predictable credential name pattern that it will save to the cred manager.

> I'm not sure (right this second) why we need this criteria cred[:target].start_with?(uri.to_s)

I fully agree and I think it's effectively doing just a bad version of `<scheme>://<host>:<port>` comparison which is the pattern we use in README but I explicitly didn't want to possibly change the behavior for current users/setups.

Reply:
I think we should create an issue to make this more robust. It applies to any scenario where credentials are committed on the agent coming from the puppet server (including HFTs)